test/integration/src/test/java/io/pravega/test/integration/BatchClientTest.java

```diff
@@ -173,6 +178,38 @@ public void testBatchClient() throws Exception {
         ArrayList<String> dataAtOffset = Lists.newArrayList(seg0Iterator);
         assertEquals(1, dataAtOffset.size());
         assertEquals(DATA_OF_SIZE_30, dataAtOffset.get(0));
+
+        //Test with Stream Truncation.
+        //1. create a StreamCut after 2 events. (offset = 2 * 30 = 60)
+        StreamCut streamCut60L = new StreamCutImpl(Stream.of(SCOPE, STREAM), ImmutableMap.of(new Segment(SCOPE, STREAM, 0), 60L));
+        //2. truncate stream
+        assertTrue("truncate stream", controller.truncateStream(SCOPE, STREAM, streamCut60L).join());
+        //3. fetch Segments
+        ArrayList<SegmentRange> segmentsPostTruncation = Lists.newArrayList(batchClient.getSegments(Stream.of(SCOPE, STREAM), StreamCut.UNBOUNDED, StreamCut.UNBOUNDED).getIterator());
+        //expected segments = 1+ 3 + 2 = 6
+        assertEquals("Expected number of segments post truncation", 6, segmentsPostTruncation.size());
+        List<String> eventsPostTruncation = new ArrayList<>();
+        segmentsPostTruncation.forEach(segInfo -> {
+            @Cleanup
+            SegmentIterator<String> segmentIterator = batchClient.readSegment(segInfo, serializer);
+            eventsPostTruncation.addAll(Lists.newArrayList(segmentIterator));
+        });
+        assertEquals("Event count post truncation", 7, eventsPostTruncation.size());
+
+        //Test with truncation post fetching segments from batch client.
+        //1. create a StreamCut at the end of segment 0 ( offset = 3 * 30 = 90)
+        StreamCut streamCut90L = new StreamCutImpl(Stream.of(SCOPE, STREAM), ImmutableMap.of(new Segment(SCOPE, STREAM, 0), 90L));
+        //2. truncate stream.
+        assertTrue("truncate stream", controller.truncateStream(SCOPE, STREAM, streamCut90L).join());
+        //3. use SegmentRange obtained before truncation.
+        SegmentRange s0 = segmentsPostTruncation.stream().filter(segmentRange -> segmentRange.getSegmentNumber() == 0).findFirst().get();
+        //4. Read non existent segment.
+        assertThrows(TruncatedDataException.class, () -> {
+            List<String> eventList = new ArrayList<>();
+            @Cleanup
+            SegmentIterator<String> segmentIterator = batchClient.readSegment(s0, serializer);
+            eventList.addAll(Lists.newArrayList(segmentIterator));
+        });
     }
 
```

Comment:
I would suggest to create an additional test method to reproduce as closely as possible the original problem and check that it has been solved (i.e., truncate an active segment, use the `getHeadStreamCut` call in `BatchClient`, and verify the number of events read after truncation). If you decide to do this, the original code that raised the issue may help you and it can be found [here](https://github.com/RaulGracia/pravega/blob/1395f09f905a79ebd848b49a0941a782465e89d1/test/system/src/test/java/io/pravega/test/system/BatchClientSimpleTest.java#L208).

Reply:
done, used getStreamInfo api in the test case.